File: 0202.js
function romanToInt(s) {
    const romanNum = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000};
    let result = 0;
    let prevValue = 0;

    for (let i = s.length - 1; i >= 0; i--) {
        const char = s[i];
        const value = romanNum[char];
        result += (value < prevValue) ? -value : value;
        prevValue = value;
    }

    return result;
}
  
  console.log(romanToInt("III"));     // Output: 3
  console.log(romanToInt("LVIII"));   // Output: 58
  console.log(romanToInt("MCMXCIV")); // Output: 1994

File: Ruby/0202.rb
def roman_to_int(s)
  roman_numerals = {'I'=>1, 'V'=>5, 'X'=>10, 'L'=>50, 'C'=>100, 'D'=>500, 'M'=>1000}
  result = 0
  prev_value = 0

  s.chars.reverse_each do |char|
    value = roman_numerals[char]
    result += (value < prev_value) ? -value : value
    prev_value = value
  end

  result
end

puts roman_to_int("XL")    # Output: 40
puts roman_to_int("LVIII")  # Output: 58
puts roman_to_int("MCMLXXXVIV")# Output: 1994
